src/v/cluster/health_monitor_types.cc

```diff
@@ -252,16 +217,12 @@ adl<cluster::node_health_report>::from(iobuf_parser& p) {
       "cluster::node_health_report", p);
 
     auto id = adl<model::node_id>{}.from(p);
-    auto redpanda_version = adl<cluster::application_version>{}.from(p);
-    auto uptime = adl<std::chrono::milliseconds>{}.from(p);
-    auto disk_space = adl<std::vector<cluster::node_disk_space>>{}.from(p);
+    auto local = adl<cluster::node::local_state>{}.from(p);
```

Comment:
There is a lot going on in this commit and it would be easier to review if it were split up. Roughly it looks like there are the following distinct changes:

1. refactoring/splitting `void health_monitor_backend::tick()`
2. pure code movement (e.g. `node_disk_space`)
3. renames (e.g. roughly `node_disk_space` to `node::disk`?)
4. introduction of local_state
5. replacement of per-field serialization with local_state serialization

a good strategy is often to introduce changes from least to most dependent (e.g. introduce types then integrate types..) but its dependent on the situation.

Reply:
Thanks. I totally get what you are saying.. When I have a better idea what I'm doing I split stuff up on the way. In this case I fumbled to this point and go back and further split this patch out of laziness as I thought this was "clear enough" / well checked by compiler. I'm happy to do that though.